envoy/common/execution_context.h

```diff
@@ -15,13 +16,42 @@ namespace Envoy {
 static constexpr absl::string_view kConnectionExecutionContextFilterStateName =
     "envoy.network.connection_execution_context";
 
+namespace Http {
+struct FilterContext;
+}
+
+namespace Tracing {
+class Span;
+}
+
 class ScopedExecutionContext;
 
 // ExecutionContext can be inherited by subclasses to represent arbitrary information associated
 // with the execution of a piece of code. activate/deactivate are called when the said execution
 // starts/ends. For an example usage, please see
 // https://github.com/envoyproxy/envoy/issues/32012.
 class ExecutionContext : public StreamInfo::FilterState::Object, NonCopyable {
+public:
+  static void setEnabled(bool value) { enabled().store(value, std::memory_order_relaxed); }
+
+  static bool isEnabled() { return enabled().load(std::memory_order_relaxed); }
+
+  static ExecutionContext* fromStreamInfo(OptRef<const StreamInfo::StreamInfo> info) {
+    if (!isEnabled() || !info.has_value()) {
+      return nullptr;
+    }
+    const auto* const_context = info->filterState().getDataReadOnly<ExecutionContext>(
+        kConnectionExecutionContextFilterStateName);
+    return const_cast<ExecutionContext*>(const_context);
+  }
+
+  // Called when enters a scope in which |span| is active.
+  // Returns an object that can do some cleanup when exits the scope.
+  virtual Envoy::Cleanup onScopeEnter(Envoy::Tracing::Span& span) PURE;
```

Comment:
I will perfer to let the `onScopeEntrer` implemetation self no handle the nullptr case.

Reply:
SG, I can do that after we settle on the `onScopeEnter` api, in the discussion below.

Reply:
Changed both `onScopeEntrer` to accept a pointer.